Microsoft.Toolkit.Uwp.UI.Controls/SwitchPresenter/SwitchPresenter.cs

```diff
@@ -0,0 +1,246 @@
+        /// </summary>
+        public static readonly DependencyProperty DataTypeProperty =
+            DependencyProperty.Register(nameof(TargetType), typeof(Type), typeof(SwitchPresenter), new PropertyMetadata(null));
+
+        /// <summary>
+        /// Gets or sets a value indicating whether the content is removed from the visual tree when switching between cases.
+        /// </summary>
+        public bool IsVisualTreeDisconnectedOnChange { get; set; }
+
+        private static void OnSwitchCasesPropertyChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            if (e.OldValue != null)
+            {
+                ((SwitchPresenter)e.OldValue).SwitchCases.CaseCollectionChanged -= OnCaseValuePropertyChanged;
+            }
+
+            var xswitch = (SwitchPresenter)d;
+
+            foreach (var xcase in xswitch.SwitchCases)
+            {
+                // Set our parent
+                xcase.Parent = xswitch;
+            }
+
+            // Will trigger on collection change and case value changed
+            xswitch.SwitchCases.Parent = xswitch;
+            xswitch.SwitchCases.CaseCollectionChanged += OnCaseValuePropertyChanged;
+        }
+
+        private static void OnValuePropertyChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            // When our Switch's expression changes, re-evaluate.
+            var xswitch = (SwitchPresenter)d;
+
+            xswitch.EvaluateCases();
+        }
+
+        private static void OnCaseValuePropertyChanged(object sender, EventArgs e)
+        {
+            // When something about our collection of cases changes, re-evaluate.
+            var collection = (CaseCollection)sender;
+
+            collection.Parent.EvaluateCases();
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SwitchPresenter"/> class.
+        /// </summary>
+        public SwitchPresenter()
+        {
+            this.SwitchCases = new CaseCollection();
+        }
+
+        /// <inheritdoc/>
+        protected override void OnApplyTemplate()
+        {
+            base.OnApplyTemplate();
+
+            EvaluateCases();
+        }
+
+        private void EvaluateCases()
+        {
+            if (CurrentCase != null &&
+                CurrentCase.Value != null &&
+                CurrentCase.Value.Equals(Value))
+            {
+                // If the current case we're on already matches our current value,
+                // then we don't have any work to do.
+                return;
+            }
+
+            Case xdefault = null;
+            Case newcase = null;
+
+            foreach (var xcase in SwitchCases)
+            {
+                if (xcase.IsDefault)
+                {
+                    // If there are multiple default cases provided, this'll override and just grab the last one, the developer will have to fix this in their XAML. We call this out in the case comments.
```

Comment:
```suggestion
                    // If there are multiple default cases provided, this will override and just grab the last one, the developer will have to fix this in their XAML. We call this out in the case comments.
```